tests/contrib/langchain/test_langchain.py

```diff
@@ -1265,7 +1266,34 @@ def test_vectorstore_logs_error(langchain, ddtrace_config_langchain, mock_logs,
 )
 class TestLLMObsLangchain:
     @staticmethod
-    def _expected_llmobs_calls(provider, span, input_role=None, output_role=None):
+    def _expected_llmobs_calls(trace, expected_spans_data: list):
```

Comment:
Just a suggestion, seems like the changing the framework for testing llm/chat_models to generalize to both spans/chain traces was not trivial and the readability has suffered a bit as a result. I would recommend leaving the original `_expected_llmobs_calls()` and `_test_llmobs_invoke()` alone to deal with single llm/chat_model test cases, and create separate helper methods (i.e. `_expected_llmobs_chain_calls()`, `_test_llmobs_invoke_chain()` to deal with the chain/workflow test cases.